src/Backups/BackupsWorker.cpp

```diff
@@ -218,9 +218,36 @@ namespace
 }
 
 
-BackupsWorker::BackupsWorker(ContextPtr global_context, size_t num_backup_threads, size_t num_restore_threads, bool allow_concurrent_backups_, bool allow_concurrent_restores_)
-    : backups_thread_pool(std::make_unique<ThreadPool>(CurrentMetrics::BackupsThreads, CurrentMetrics::BackupsThreadsActive, num_backup_threads, /* max_free_threads = */ 0, num_backup_threads))
-    , restores_thread_pool(std::make_unique<ThreadPool>(CurrentMetrics::RestoreThreads, CurrentMetrics::RestoreThreadsActive, num_restore_threads, /* max_free_threads = */ 0, num_restore_threads))
+BackupsWorker::BackupsWorker(
+    ContextPtr global_context,
+    size_t num_backup_threads,
+    size_t num_restore_threads,
+    bool allow_concurrent_backups_,
+    bool allow_concurrent_restores_)
+    : backups_thread_pool(std::make_unique<ThreadPool>(
+        CurrentMetrics::BackupsThreads,
+        CurrentMetrics::BackupsThreadsActive,
+        num_backup_threads,
+        /* max_free_threads = */ 0,
+        num_backup_threads))
+    , restores_thread_pool(std::make_unique<ThreadPool>(
+          CurrentMetrics::RestoreThreads,
+          CurrentMetrics::RestoreThreadsActive,
+          num_restore_threads,
+          /* max_free_threads = */ 0,
+          num_restore_threads))
+    , backup_async_executor_pool(std::make_unique<ThreadPool>(
+          CurrentMetrics::BackupsThreads,
+          CurrentMetrics::BackupsThreadsActive,
+          num_backup_threads,
+          num_backup_threads,
```

Comment:
The sizes of these aux pools are not important since we allow them to free all threads.